src/simulation/power/simpleBattery/_UnitTest/test_simpleBatteryFault.py

```diff
@@ -0,0 +1,119 @@
+#
+#  ISC License
+#
+#  Copyright (c) 2016, Autonomous Vehicle Systems Lab, University of Colorado at Boulder
+#
+#  Permission to use, copy, modify, and/or distribute this software for any
+#  purpose with or without fee is hereby granted, provided that the above
+#  copyright notice and this permission notice appear in all copies.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+#  WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+#  MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+#  ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+#  WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+#  ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+#  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+#
+
+import inspect
+import os
+import pytest
+import numpy as np
+
+filename = inspect.getframeinfo(inspect.currentframe()).filename
+path = os.path.dirname(os.path.abspath(filename))
+bskName = 'Basilisk'
+splitPath = path.split(bskName)
+
+from Basilisk.utilities import SimulationBaseClass
+from Basilisk.architecture import messaging
+from Basilisk.simulation import simpleBattery
+from Basilisk.utilities import macros
+
+params_set_data = [(5., 5., 5., 10., 0.3),
+                    (1., 1., 5., 10., 0.3),
+                    (5, 5, 5, 10, 1e-3),
+                    (5., -5., 5., 10., 0.3),
+                    (5., 5., 5., 10., -0.3),
+                    (5., 5., 5., 10., 1.3)]
+
+@pytest.mark.parametrize(
+        "storedChargeInit, netPower_1, netPower_2, batteryCapacity, faultCapacityRatio",
+        params_set_data)
+
+def test_storage_limits(storedChargeInit, netPower_1, netPower_2, batteryCapacity, faultCapacityRatio):
+    """
+    **Validation Test Description**
+    1. Check if the battery capacity matches its nominal value when a fault occurs.
+    2. Verify that the charged capacity does not exceed the faulted capacity or drop below zero.
+
+    :param show_plots: Not used; no plots to be shown.
+
+    :return:
+    """
+
+    unitTaskName = "unitTask"               # arbitrary name (don't change)
+    unitProcessName = "TestProcess"         # arbitrary name (don't change)
+
+    # Create a sim module as an empty container
+    unitTestSim = SimulationBaseClass.SimBaseClass()
+
+    # Create test thread
+    testProcessRate = macros.sec2nano(0.1)     # update process rate update time
+    testProc = unitTestSim.CreateNewProcess(unitProcessName)
+    testProc.addTask(unitTestSim.CreateNewTask(unitTaskName, testProcessRate))
+
+    test_battery = simpleBattery.SimpleBattery()
+    test_battery.storedCharge_Init = storedChargeInit
+    test_battery.storageCapacity = batteryCapacity  
+    test_battery.fault = True
```

Comment:
Right now you are only testing the functionality if there is a fault.  You should include a trivial test that checks that the default behavior is no fault.